### PR TITLE
Deprecate KFP preceding component

### PR DESCRIPTION
**Pull request by cloudw**
Remove KFP preceding component from code. Also removing related test, example flows and doc strings.

### CODE REVIEW

File: metaflow/plugins/kfp/tests/flows/resources_flow.py [hunk 1]
```diff
@@ -126,6 +126,7 @@ class ResourcesFlow(FlowSpec):
     @resources(
         local_storage="242",
         cpu="0.6",
+        local_storage="242",
```
Comment (alexlatchford) [inline metaflow/plugins/kfp/tests/flows/resources_flow.py]: Looks like this is a typo?

**Review by alexlatchford — APPROVED** [review]
Looks great! Excited to slim this down, thanks for picking this up!

File: metaflow/plugins/kfp/tests/flows/resources_flow.py (hunk 1 above)
Comment (cloudw) [reply metaflow/plugins/kfp/tests/flows/resources_flow.py]: Ah not sure where that comes from. Thanks for spotting it!

**Comment by talebzeghmi** [comment]
Thank you Yun for owning this!
I suggest a simplification.  

To instead raise an exception if the user adds parameter `preceding_component, preceding_component_inputs, preceding_component_outputs`.  This is a one liner and not risky.  It also gives us the ability to add it back in the I pray very unlikely event we must add it.  We may need to add it not for customers but for say use by pipeler or LCP data contract evaluation, etc.  

The one liner gives us more flexibility going forward, and is less risky with the same result.  Maybe more than a one liner because we can remove the documentation also of parameters `preceding_component, preceding_component_inputs, preceding_component_outputs`.

**Comment by cloudw** [comment]
> To instead raise an exception if the user adds parameter `preceding_component, preceding_component_inputs, preceding_component_outputs`. This is a one liner and not risky. It also gives us the ability to add it back in the I pray very unlikely event we must add it. We may need to add it not for customers but for say use by pipeler or LCP data contract evaluation, etc.
> 
> The one liner gives us more flexibility going forward, and is less risky with the same result. Maybe more than a one liner because we can remove the documentation also of parameters `preceding_component, preceding_component_inputs, preceding_component_outputs`.

Thanks @talebzeghmi for suggesting a simpler way to deprecate. 

Could you elaborate on why KFP preceding_component 
would be required for pipeler or LCP data contract evaluation, and the same couldn't be accomplished by only using metaflow?

**Comment by talebzeghmi** [comment]
>Could you elaborate on why KFP preceding_component
would be required for pipeler or LCP data contract evaluation, and the same couldn't be accomplished by only using metaflow?

`preceding_component` gives us the ability to interject KFP component steps in between Metaflow steps.  This feature could be useful if we keep the KFP plugin.

**Review by talebzeghmi — APPROVED** [review]

**Pull request merged**